src/DevBetterWeb.Web/Pages/Videos/Details.cshtml.cs

```diff
@@ -20,73 +24,137 @@ namespace DevBetterWeb.Web.Pages.Videos;
 [Authorize(Roles = AuthConstants.Roles.ADMINISTRATORS_MEMBERS_ALUMNI)]
 public class DetailsModel : PageModel
 {
-  [BindProperty]
-  public OEmbedViewModel? OEmbedViewModel { get; set; }
-
-  private readonly GetOEmbedVideoService _getOEmbedVideoService;
-  private readonly GetVideoService _getVideoService;
-  private readonly IRepository<ArchiveVideo> _repository;
-  private readonly IMapper _mapper;
-  private readonly IMarkdownService _markdownService;
-  private readonly UserManager<ApplicationUser> _userManager;
-  private readonly IRepository<Member> _memberRepository;
-
-  public DetailsModel(IMapper mapper, IMarkdownService markdownService, GetOEmbedVideoService getOEmbedVideoService, 
-	  GetVideoService getVideoService, IRepository<ArchiveVideo> repository, UserManager<ApplicationUser> userManager, 
-	  IRepository<Member> memberRepository)
-  {
-	  _mapper = mapper;
-	  _markdownService = markdownService;
-    _getVideoService = getVideoService;
-    _repository = repository;
-    _getOEmbedVideoService = getOEmbedVideoService;
-    _userManager = userManager;
-    _memberRepository = memberRepository;
-  }
+	[BindProperty]
+	public OEmbedViewModel? OEmbedViewModel { get; set; }
+	public string? Transcript { get; set; }
 
-  public async Task<IActionResult> OnGet(string videoId, string? startTime = null)
-  {
-    var video = await _getVideoService.ExecuteAsync(videoId);
-    if (video?.Data == null) return NotFound($"Video Not Found {videoId}");
+	private readonly GetOEmbedVideoService _getOEmbedVideoService;
+	private readonly GetVideoService _getVideoService;
+	private readonly IRepository<ArchiveVideo> _repository;
+	private readonly IMapper _mapper;
+	private readonly IMarkdownService _markdownService;
+	private readonly UserManager<ApplicationUser> _userManager;
+	private readonly IRepository<Member> _memberRepository;
+	private readonly GetAllTextTracksService _getAllTextTracksService;
+	private readonly HttpClient _httpClient;
+
+	public DetailsModel(IMapper mapper, IMarkdownService markdownService, GetOEmbedVideoService getOEmbedVideoService,
+		GetVideoService getVideoService, IRepository<ArchiveVideo> repository, UserManager<ApplicationUser> userManager,
+		IRepository<Member> memberRepository, GetAllTextTracksService getAllTextTracksService, HttpClient httpClient)
+	{
+		_mapper = mapper;
+		_markdownService = markdownService;
+		_getVideoService = getVideoService;
+		_repository = repository;
+		_getOEmbedVideoService = getOEmbedVideoService;
+		_userManager = userManager;
+		_memberRepository = memberRepository;
+		_getAllTextTracksService = getAllTextTracksService;
+		_httpClient = httpClient;
+	}
```

Comment:
I was hoping the 7 constructor dependencies that were here would move closer to 1, not increase to 9, as part of the refactoring to move some of this into a separate service.

Reply:
Oh yeah, I'm definitely going to improve that. This will involve creating 1 or more classes. I didn't want to do to much in this one PR. I focused on refactoring at the method level. Next step, classes.